### ISSUE TITLE
PARAGRAPH span must start at paragraph boundary (no linebreak in li tags possible)

### ISSUE THREAD

**Issue by msdx** [sufficientlysecure/html-textview#6]
```
java.lang.RuntimeException: PARAGRAPH span must start at paragraph boundary
            at android.text.SpannableStringBuilder.setSpan(SpannableStringBuilder.java:604)
            at android.text.SpannableStringBuilder.setSpan(SpannableStringBuilder.java:592)
            at android.text.HtmlToSpannedConverter.convert(Html.java:461)
            at android.text.Html.fromHtml(Html.java:138)
            at org.sufficientlysecure.htmltextview.HtmlTextView.setHtmlFromString(HtmlTextView.java:82)
```

**Comment by dschuermann** [comment]
To analyze this, we need the HTML producing this error.

**Comment by tyk5555** [comment]
I have the same issue too, using the latest version.
Here is the HTML:

```html
<html>
 <head></head>
 <body>
  HAL向印军交付首架自修苏30 2015-01-12 09:01 
  <img src="http://y0.ifengimg.com/2f86f1d4ae63c9a4/2012/0522/icon_07.gif" alt="">
  <a href="http://comment.ifeng.com/view.php?chId=17667&amp;docId=42911815&amp;docName=HAL%E5%90%91%E5%8D%B0%E5%86%9B%E4%BA%A4%E4%BB%98%E9%A6%96%E6%9E%B6%E8%87%AA%E4%BF%AE%E8%8B%8F30%20%E8%A2%AB%E7%A7%B0%E9%A3%9E%E8%A1%8C%E6%A3%BA%E6%9D%90%E5%88%B6%E9%80%A0%E8%80%85&amp;docUrl=http://news.ifeng.com/a/20150112/42911815_0.shtml&amp;skey=9b5b33" rel="nofollow"><i>107</i>人参与</a>
  <a href="http://comment.ifeng.com/view.php?chId=17667&amp;docId=42911815&amp;docName=HAL%E5%90%91%E5%8D%B0%E5%86%9B%E4%BA%A4%E4%BB%98%E9%A6%96%E6%9E%B6%E8%87%AA%E4%BF%AE%E8%8B%8F30%20%E8%A2%AB%E7%A7%B0%E9%A3%9E%E8%A1%8C%E6%A3%BA%E6%9D%90%E5%88%B6%E9%80%A0%E8%80%85&amp;docUrl=http://news.ifeng.com/a/20150112/42911815_0.shtml&amp;skey=9b5b33" rel="nofollow"><i>75</i>条评论</a>
  <br> 
  <br> 
  <img src="http://y0.ifengimg.com/a/2015_03/8b3846dbe559114.jpg" alt="">
  <br> 
  <br> [url=]
  <img src="http://y0.ifengimg.com/a/2015_03/8b3846dbe559114.jpg" alt="">[/url]
  <br> 
  <br> 
  <br> 
  <ul>
   <li><a href="http://news.ifeng.com/a/20150112/42911815_0.shtml#p=1" rel="nofollow"><img src="http://d.ifengimg.com/w179_h150/y0.ifengimg.com/a/2015_03/8b3846dbe559114.jpg" alt=""></a></li>
   <li><a href="http://news.ifeng.com/a/20150112/42911815_0.shtml#p=2" rel="nofollow"><img src="http://d.ifengimg.com/w179_h150/y1.ifengimg.com/a/2015_03/415365aa3cbf449.jpg" alt=""></a></li>
   <li><a href="http://news.ifeng.com/a/20150112/42911815_0.shtml#p=3" rel="nofollow"><img src="http://d.ifengimg.com/w179_h150/y2.ifengimg.com/a/2015_03/af11f43d4827e87.jpg" alt=""></a></li>
   <li><a href="http://news.ifeng.com/a/20150112/42911815_0.shtml#p=4" rel="nofollow"><img src="http://d.ifengimg.com/w179_h150/y1.ifengimg.com/a/2015_03/1243f59d4e89d5a.jpg" alt=""></a></li>
   <li><a href="http://news.ifeng.com/a/20150112/42911815_0.shtml#p=5" rel="nofollow"><img src="http://d.ifengimg.com/w179_h150/y3.ifengimg.com/a/2015_03/c21e4316cda5320.jpg" alt=""></a></li>
   <li><a href="http://news.ifeng.com/a/20150112/42911815_0.shtml#p=6" rel="nofollow"><img src="http://d.ifengimg.com/w179_h150/y2.ifengimg.com/a/2015_03/effca48cfa08d65.jpg" alt=""></a></li>
   <li><a href="http://news.ifeng.com/a/20150112/42911815_0.shtml#p=7" rel="nofollow"><img src="http://d.ifengimg.com/w179_h150/y3.ifengimg.com/a/2015_03/32ae13ea67f7e76.jpg" alt=""></a></li>
   <li><a href="http://news.ifeng.com/a/20150112/42911815_0.shtml#p=8" rel="nofollow"><img src="http://d.ifengimg.com/w179_h150/y2.ifengimg.com/a/2015_03/4fccf5eb265127a.jpg" alt=""></a></li>
   <li><a href="http://news.ifeng.com/a/20150112/42911815_0.shtml#p=9" rel="nofollow"><img src="http://d.ifengimg.com/w179_h150/y3.ifengimg.com/a/2015_03/167557afd65214e.jpg" alt=""></a><br> </li>
  </ul>
  <br> 
  <br> 
  <br> 
  <br> 
  <br> 
  <br> 
  <br> 
  <br> 
  <br> 
  <p>新年伊始，印度又有喜讯传来。1月9日，印度斯坦航空公司（HAL）向印度空军交付了该公司修复的首架苏-30MKI战机。这也意味着，印度HAL公司全面掌握了维修苏-30战机的能力。这架苏-30战机印度空军编号SB-027，由俄罗斯伊尔库特航空公司制造，2002年9月16日交付印度空军。印度国防部长巴里卡和印度空军总司令阿鲁普·拉哈出席交付仪式。印度斯坦航空公司)是印度国防部下辖的国防生产部直属的九大国营企业之中最大的一家企业，也是印度唯一的航空制造公司，公司下辖12个制造厂，但作为“整体萎靡”的印度国防工业的一个缩影</p>
  <br> 
  <br> 
  <p>[url=]<img src="http://y1.ifengimg.com/a/2015_03/415365aa3cbf449.jpg" alt="">[/url]</p>
  <br> 
  <br> [p=20, null, left]1992-2004年间, HAL一共组装米格-21两架,大修八架, 其中八架坠毁；组装米格-21BISON一架,大修一架,其中一架坠毁；组装美洲豹三架,大修五架,其中六架坠毁；大修幻影-2000四架,全部坠毁；组装米格-27三架,大修三架,其中四架坠毁；大修米格-29三架,全部坠毁；2004年至2007年间，印军坠毁的29架飞机中，有26架是由HAL制造。
  <p></p>
  <br> 
  <p><br> </p>
  <p>++</p>
  <p><br> </p>
  <p><br> </p>
  <p><br> [url=]<img src="http://y2.ifengimg.com/a/2015_03/af11f43d4827e87.jpg" alt="">[/url]</p>
  <br> 
  <br> [p=20, null, left]据印度媒体报道，2014年10月14日17时30分，印度空军一架国旗涂装苏-30MKI战机失事坠落。在事发地在距离印度普纳市25公里处，大量当地民众前来围观。相关人士称，此次事故并未造成人员伤亡。这是继2009年4月30日、2009年11月30日、2011年12月13日和2013年2月19日后，印度空军苏-30MKI战机损失的第5架机。值得注意的是，2011年苏-30坠毁事件也发生在普纳附近。
  <p></p>
  <br> 
  <p><br> </p>
  <p><br> </p>
  <p><br> [url=]<img src="http://y1.ifengimg.com/a/2015_03/1243f59d4e89d5a.jpg" alt="">[/url]</p>
  <br> 
  <br> [p=20, null, left]据印度媒体报道，2014年10月14日17时30分，印度空军一架国旗涂装苏-30MKI战机失事坠落。在事发地在距离印度普纳市25公里处，大量当地民众前来围观。相关人士称，此次事故并未造成人员伤亡。这是继2009年4月30日、2009年11月30日、2011年12月13日和2013年2月19日后，印度空军苏-30MKI战机损失的第5架机。值得注意的是，2011年苏-30坠毁事件也发生在普纳附近。
  <p></p>
  <br> 
  <p><br> </p>
  <p><br> </p>
  <p><br> </p>
  <p><br> </p>
  <p><br> </p>
  <br> 
  <br> 
 </body>
</html>
```
Thank you very much.

**Comment by zuoyanyouwu** [comment]
The HTML like this:
```
 <ul><li><a href="http://www.zhihu.com/people/chen-yu-lan-25">ChenYuLan</a>：A Doctor。<br/></li></ul>
```
Application would throws that exception .
If I delete `</br>` , the problem solved.

**Comment by tyk5555** [comment]
@zuoyanyouwu Thank you very much, the problem is solved!

**Comment by msdx** [comment]
@dschuermann I have forgetten it since I havn't pay attention this issue for a long time. I think the HTML may be some section of this:http://gradledoc.qiniudn.com/1.12/userguide/userguide.html.
I think the cause  is same as above.

**Comment by dschuermann** [comment]
Should now be fixed in 1.1. See https://github.com/sufficientlysecure/html-textview#how-to-import how to get this version!
